File: db/src/main/java/org/delivery/db/storeuser/StoreUserRepository.java
package org.delivery.db.storeuser;

import java.util.Optional;
import org.delivery.db.storeuser.enums.StoreUserStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StoreUserRepository extends JpaRepository<StoreUserEntity, Long> {

    // 이메일과 상태로 가게 사용자 엔티티 하나 찾기
    // select * from store_user where email = ? and status = ? order by id desc limit 1
    Optional<StoreUserEntity> findFirstByEmailAndStatusOrderByIdDesc(String email, StoreUserStatus status);
}

File: db/src/main/kotlin/org/delivery/db/storeuser/StoreUserRepository.kt
package org.delivery.db.storeuser

import org.delivery.db.storeuser.enums.StoreUserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreUserRepository : JpaRepository<StoreUserEntity, Long> {
    // 이메일과 상태로 가게 사용자 엔티티 하나 찾기
    // select * from store_user where email = ? and status = ? order by id desc limit 1
    fun findFirstByEmailAndStatusOrderByIdDesc(email: String?, status: StoreUserStatus?): StoreUserEntity?
}
